src/main/java/sqltoregex/settings/SettingsManager.java

```diff
@@ -55,26 +58,57 @@ public <S> Set<RegExGenerator<S>> getSettingByClass(Class<? extends RegExGenerat
     }
 
     public boolean getSettingBySettingsOption(SettingsOption settingsOption) {
-        return this.settingsMap.containsKey(settingsOption);
+        return this.getSettingBySettingsOption(settingsOption, false);
+    }
+
+    public boolean getSettingBySettingsOption(SettingsOption settingsOption, boolean useDefault) {
+        if (useDefault){
+            return this.getDefaultSettingsMap().containsKey(settingsOption);
+        }else {
+            return this.getSettingsMap().containsKey(settingsOption);
+        }
     }
 
     public <S> RegExGenerator<S> getSettingBySettingsOption(SettingsOption settingsOption,
                                                             Class<? extends RegExGenerator<S>> clazz) {
-        for (Map.Entry<SettingsOption, RegExGenerator<?>> entry : this.settingsMap.entrySet()) {
+        return this.getSettingBySettingsOption(settingsOption, clazz, false);
```

Comment:
if u set USE_DEFAULT for true, why not a better readable variable for false?

Reply:
USE_DEFAULT ist used outside of Settings_Manager to increase readability and self documentation. Inside the class the assumption is made, that someone knows how the class is built up and therefore there is no constant synonym used.
@binkertpat